src/rocksdb.rs

```diff
@@ -39,16 +40,134 @@ pub struct ReadOptions {
     inner: rocksdb_ffi::RocksDBReadOptions,
 }
 
+pub struct Snapshot<'a> {
+    db: &'a RocksDB,
+    inner: rocksdb_ffi::RocksDBSnapshot,
+}
+
+pub struct DBIterator {
+//TODO: should have a reference to DB to enforce scope, but it's trickier than I thought to add
+    inner: rocksdb_ffi::RocksDBIterator,
+    direction: Direction,
+    just_seeked: bool
+}
+
+pub enum Direction {
+  forward, reverse
+}
```

Comment:
Is there a reason why you implemented SubDBIterator instead of [DoubleEndedIterator](https://doc.rust-lang.org/std/iter/trait.DoubleEndedIterator.html)?